src/util/system.cpp

```diff
@@ -428,6 +428,21 @@ bool ArgsManager::ReadSettingsFile(std::vector<std::string>* errors)
         SaveErrors(read_errors, errors);
         return false;
     }
+    for (const auto& setting : m_settings.rw_settings) {
+        std::string section;
+        std::string key = setting.first;
+        (void)InterpretOption(section, key, /* value */ {}); // Split setting key into section and argname
+        Optional<unsigned int> flags = GetArgFlags('-' + key);
+        if (flags) {
+            std::string val_error;
+            if (!CheckValid(key, /* value */ setting.second, *flags, val_error)) {
+                SaveErrors({val_error}, errors);
+                return false;
+            }
+        } else {
+            LogPrintf("Ignoring unknown rw_settings value %s\n", setting.first);
+        }
```

Comment:
In commit "Interpret and validate rw_settings" (faec629663a92f28c5328ab3877ff58847bf9e35)

Would suggest just:

```c++
if (!flags) {
    LogPrintf("Ignoring unknown rw_settings value %s\n", setting.first);
}
```

I don't think this can return false and prevent loading all settings when one has an unrecognized value, because it could break downgrading if allowed values are expanded, similar to what's described https://github.com/bitcoin/bitcoin/pull/15935#issuecomment-663566209. I also tend to think there shouldn't call a new call to CheckValid while that function is basically broken and its behavior is being worked out in other PRs.

Reply:
Ok, removed `CheckValid` for now